// See README.txt for information and build instructions.

package pb;


// 格式 S_服务器类型_TO_服务器类型_消息号类型，如下解释：
// S_ 开头，服务器消息号。
// G : 表游戏服务器
// C : 表中心服务器
// L : 表登录服务器
// D : 数据库服务器
// E : 事件服务器
// B : 房间匹配服务器
// R : 跨服战斗服务器
enum ServerOpcode
{
////////////////////////////////////////////////////////////////////////////////

	S_PING                                            =1;
	S_PONG                                            =2;
////////////////////////////////////////////////////////////////////////////////
//GameToCenter
	SG2C_REGISTER                                     =3;
	SG2C_REGISTER_RET                                 =4;
	SG2C_UPDATE_INFO                                  =5;

	SG2C_GET_LOGIN_SERVERS                            =6;
	SG2C_GET_LOGIN_SERVERS_RET                        =7;

	SG2C_GET_CROSS_SERVERS                            =8;
	SG2C_GET_CROSS_SERVERS_RET                        =9;

	SG2C_BROADCAST_TO_LOGIN_SERVER                    =10;

	SG2C_GET_EVENT_LOG_SERVER                         =11;
	SG2C_GET_EVENT_LOG_SERVER_RET                     =12;

	SG2C_EXCHANGE_GIFT_CODE                           =13;
	SG2C_EXCHANGE_GIFT_CODE_RET                       =14;

	SG2C_FEEDBACK                                     =15;
	SG2C_FEEDBACK_RET                                 =16;

////////////////////////////////////////////////////////////////////////////////
//GameToLogin
	SG2L_REGISTER                                     =17;
	SG2L_REGISTER_RET                                 =18;
	SG2L_BROADCAST                                    =19;

//GameToLobby
	SG2B_REGISTER                                     =20;
	SG2B_REGISTER_RET                                 =21;

//GameToCross
	SG2R_REGISTER                                     =22;
	SG2R_REGISTER_RET                                 =23;

////////////////////////////////////////////////////////////////////////////////
//GameToEventLog
	SG2E_REGISTER                                     =24;
	SG2E_REGISTER_RET                                 =25;
	SG2E_EVENT                                        =26;

////////////////////////////////////////////////////////////////////////////////
//CenteroGame
	SC2G_ADD_LOGIN_SERVERS                            =27;

	SC2G_ADD_EVENT_LOG_SERVER                         =28;

	SC2G_ADD_CROSS_SERVERS                            =29;

	SC2G_WEB_GET_PLAYER_LIST                          =30;
	SC2G_WEB_GET_PLAYER_LIST_RET                      =31;

	SC2G_WEB_PAY                                      =32;
	SC2G_WEB_PAY_RET                                  =33;

	SC2G_SEND_SERVERS_MAIL                            =34;
	SC2G_SEND_PLAYER_MAIL                             =35;
	SC2G_SEND_PLAYER_MAIL_RET                         =36;
	SC2G_PLAYER_FORBID                                =37;
	SC2G_PLAYER_FORBID_RET                            =38;
	SC2G_SET_PLAYER_VIP                               =39;
	SC2G_SET_PLAYER_VIP_RET                           =40;
	SC2G_RECREATE_PAY_ORDER                           =41;
	SC2G_RECREATE_PAY_ORDER_RET                       =42;
	SC2G_SEND_SERVERS_NOTICE                          =43;
	SC2G_SERVERS_HOT_RES_RELOAD                       =44;
	SC2G_GET_ONLINE_PLAYER_COUNT                      =45;
	SG2C_GET_ONLINE_PLAYER_COUNT_RET                  =46;
	SC2G_GET_SERVER_INFO                              =47;//
	SG2C_GET_SERVER_INFO_RET                          =48;//
	SC2G_SERVERS_HOT_PAY_ITEM_RELOAD                  =49;
	SC2G_SERVERS_SAVE_DBPROXY                         =50;


////////////////////////////////////////////////////////////////////////////////
//CenterToLogin
	SC2L_BROADCAST                                    =51;

	SC2L_ADD_EVENT_LOG_SERVER                         =52;

//CenterToLobby
	SC2B_ADD_EVENT_LOG_SERVER                         =53;

//CenterToCross
	SC2R_ADD_EVENT_LOG_SERVER                         =54;

////////////////////////////////////////////////////////////////////////////////
//CrossToCenter
	SR2C_REGISTER                                     =55;
	SR2C_REGISTER_RET                                 =56;
	SR2C_GET_LOBBY_SERVER                             =57;
	SR2C_GET_LOBBY_SERVER_RET                         =58;
	SR2C_GET_EVENT_LOG_SERVER                         =59;
	SR2C_GET_EVENT_LOG_SERVER_RET                     =60;
	SR2C_GET_GAME_SERVERS                             =61;
	SR2C_GET_GAME_SERVERS_RET                         =62;

////////////////////////////////////////////////////////////////////////////////
//CrossToLobby
	SR2B_REGISTER                                     =63;
	SR2B_REGISTER_RET                                 =64;

////////////////////////////////////////////////////////////////////////////////
//CrossToGame
	SR2G_REGISTER                                     =65;
	SR2G_REGISTER_RET                                 =66;

////////////////////////////////////////////////////////////////////////////////
//CrossToEventLog
	SR2E_REGISTER                                     =67;
	SR2E_REGISTER_RET                                 =68;
	SR2E_EVENT                                        =69;

////////////////////////////////////////////////////////////////////////////////
//LobbyToCenter
	SB2C_REGISTER                                     =70;
	SB2C_REGISTER_RET                                 =71;
	SB2C_GET_EVENT_LOG_SERVER                         =72;
	SB2C_GET_EVENT_LOG_SERVER_RET                     =73;

////////////////////////////////////////////////////////////////////////////////
//LobbyToEventLog
	SB2E_REGISTER                                     =74;
	SB2E_REGISTER_RET                                 =75;
	SB2E_EVENT                                        =76;


////////////////////////////////////////////////////////////////////////////////
//LoginToEventLog
	SL2E_REGISTER                                     =77;
	SL2E_REGISTER_RET                                 =78;
	SL2E_EVENT                                        =79;

////////////////////////////////////////////////////////////////////////////////
//LoginToCenter
	SL2C_REGISTER                                     =80;
	SL2C_REGISTER_RET                                 =81;

	SL2C_GET_EVENT_LOG_SERVER                         =82;
	SL2C_GET_EVENT_LOG_SERVER_RET                     =83;

////////////////////////////////////////////////////////////////////////////////
//EventLogToCenter
	SE2C_REGISTER                                     =84;
	SE2C_REGISTER_RET                                 =85;



////////////////////////////////////////////////////////////////////////////////
//PlatfromToGameserver
	SP2G_MAIL_MSG                                     =86;
//DatabaseToGame
	SD2G_LOADINGGUILDINFO                             =87;//加载公会数据
	SD2G_GUILD_SEARCH_PLAYERSINFO_RESPON              =88;//返回公会搜索玩家信息

	SD2G_GET_SLOTITEMINFO                             =89;//返回物品道具信息
	SD2G_GET_PLAYERINFO_RESPON                        =90;//返回申请玩家信息
	SD2G_IS_PLAYEREXIST_RESPON                        =91;//返回玩家是否存在的信息

	SD2G_LOADFRIENDINFO                               =92;//加载好友数据
	SD2G_SOCIAL_SEARCH_PLAYERSINFO_RESPON             =93;//返回好友搜索玩家信息

	SD2G_GET_PLAYER_APPEND_DATA_RET                   =94;

	SD2G_PLAYER_CACHE_DATA_SYNC                       =95;

	SD2G_PLAYER_CACHE_REMOVE_SYNC                     =96;

////////////////////////////////////////////////////////////////////////////////
//GameToDatabase
//                                                =120$;
	SG2D_REGISTER                                     =120;
	SD2G_REGISTER_RET                                 =121;

	SG2D_SAVE_PLAYER_CACHE                            =122;

	SG2D_GET_PLAYER_APPEND_DATA                       =123;

	SG2D_BROADCAST_OFFLINE_MSG                        =124;
	SD2G_BROADCAST_OFFLINE_RET                        =125;

	SG2D_GET_PLAYER_LIST                              =126;
	SD2G_GET_PLAYER_LIST_RET                          =127;

	SG2D_CREATE_PLAYER                                =128;
	SD2G_CREATE_PLAYER_RET                            =129;

	SG2D_SELSEC_PALYER                                =130;
	SD2G_SELSEC_PALYER_RET                            =131;

	SG2D_PLAYER_OFFLINE                               =132;
	SG2D_CREATE_PLAYER_DATA                           =133;//数据库服务器构建我玩家数据
	SG2D_UPDATA_STAGELEVEL_INFO                       =134;
	SD2G_UPDATA_STAGELEVEL_INFO_RET                   =135;

	SG2D_GET_SLOTITEMINFO                             =136;//申请获得物品道具信息
	SG2D_GET_PLAYERINFO                               =137;//申请获得玩家信息
	SG2D_IS_PLAYEREXIST                               =138;//玩家是否存在的信息

	SG2D_UPDATE_GUILD_DATA                            =139;//更新公会数据
	SG2D_UPDATE_GUILD_MEMBER_DATA                     =140;//更新公会数据
	SG2D_GUILD_SEARCH_PLAYERSINFO                     =141;//公会申请搜索玩家信息

	SG2D_CREAT_GUILD                                  =142;//创建公会
	SD2G_CREAT_GUILD_RET                              =143;//返回创建公会结果


	SG2D_UPDATE_FRIEND_DATA                           =144;//更新好友数据
	SG2D_SOCIAL_SEARCH_PLAYERSINFO                    =145;//好友申请搜索玩家信息
	SG2D_FRIEND_OPERATE                               =146;//玩家的好友操作。
	SD2G_FRIEND_OPERATE_RET                           =147;//玩家的好友操作结果。

	SG2D_SAVE_MALL_PROMOTE                            =148;

	SG2D_CHECK_MAIL_RECIVERS_EXIST                    =149;
	SD2G_CHECK_MAIL_RECIVERS_EXIST_RET                =150;

	SG2D_MAIL_RECEVE_MAIL                             =151;

	SG2D_MAIL_UPDATEMAILINFO                          =152;
	SD2G_MAIL_UPDATEMAILINFO_RET                      =153;

	SG2D_UPDATE_BAG                                   =154;
	SG2D_UPDATE_PLAYER_VALUE                          =155;//playervalueholder

	SG2D_CREATE_CHAR                                  =156;
	SG2D_UPDATE_CHAR                                  =157;
	SG2D_DELETE_CHARS                                 =158;
	SG2D_UPDATE_BATTLE_CHARS                          =159;

	SG2D_UPDATE_QUEST                                 =160;
	SG2D_COMPLETE_QUEST                               =161;
	SG2D_UPDATE_BEHAVIOR_COUNTER                      =162;

	SG2D_UPDATE_SELL_RANDOM_PROGRAME                  =163;
	SG2D_UPDATE_SELL_PROGRAM_INFO                     =164;

	SG2D_UPDATE_PLAYER_STRING                         =165;

	SG2D_UPDATE_LOOT_RECORD                           =166;
	SG2D_UPDATE_LOTTERY_RECORD                        =167;


	SG2D_WEB_GET_PLAYER_LIST                          =168;
	SD2G_WEB_GET_PLAYER_LIST_RET                      =169;

	SG2D_PVP_CREATE_VIRTUAL_PLAYER                    =170;
	SD2G_PVP_CREATE_VIRTUAL_PLAYER_RET                =171;

	SG2D_PVP_ENTRY_ARENA                              =172;
	SD2G_PVP_ENTRY_ARENA_RET                          =173;

	SG2D_PVP_ARENA_PLAYER_INFO                        =174;
	SD2G_PVP_ARENA_PLAYER_INFO_RET                    =175;

	SG2D_PVP_ARENA_FIGHT_RESULT_INFO                  =176;
	SD2G_PVP_ARENA_FIGHT_RESULT_INFO_RET              =177;

	SG2D_PVP_ARENA_FIGLE_RECORD_INFO                  =178;
	SD2G_PVP_ARENA_FIGLE_RECORD_INFO_RET              =179;

	SG2D_PLAYER_CHANGE_NAME                           =180;
	SD2G_PLAYER_CHANGE_NAME_RET                       =181;

	SG2D_PLAYER_SET_NAME_AND_HEADID                   =182;
	SD2G_PLAYER_SET_NAME_AND_HEADID_RET               =183;

	SG2D_SERVER_GOBAL_VALUE                           =184;
	SD2G_SERVER_GOBAL_VALUE_RET                       =185;

	SG2D_PVP_GET_REWARD_RANK_LIST                     =186;
	SD2G_PVP_GET_REWARD_RANK_LIST_RET                 =187;

	SG2D_PVP_ARENA_RANKING_LIST                       =188;//竞技场排名
	SD2G_PVP_ARENA_RANKING_LIST_RET                   =189;

	SG2D_UPDATE_ENDLESS_LEVEL                         =190;

	SG2D_WEB_PAY                                      =191;
	SG2D_WEB_PAY_RET                                  =192;


	SG2D_UPDATE_MODULE_RECORD                         =193;

	SG2D_UPDATE_BUILDING_RECORD                       =194;

	SG2D_UNLOCK_FORBIDEN                              =195;
	SD2G_UNLOCK_FORBIDEN_RET                          =196;

	SG2D_SAVE_DBPROXY                                 =197;

////////////////////////////////////////////////////////////////////////////////
//WebToCenter

	SW2C_GET_SERVER_LIST                              =198;//GetGameServerList
	SW2C_GET_SERVER_LIST_RET                          =199;//GameServerList

	SW2C_GET_PLAYER_LIST                              =200;//WebGetPlayerList
	SW2C_GET_PLAYER_LIST_RET                          =201;//WebGetPlayerListRet

	SW2C_PAY                                          =202;//WebPay
	SW2C_PAY_RET                                      =203;//WebPayRet

	SW2C_SEND_SERVERS_MAIL                            =204;//
	SW2C_SEND_SERVERS_MAIL_RET                        =205;//

	SW2C_SEND_PLAYER_MAIL                             =206;//
	SW2C_SEND_PLAYER_MAIL_RET                         =207;//

	SW2C_PLAYER_FORBID                                =208;//
	SW2C_PLAYER_FORBID_RET                            =209;//

	SW2C_SET_PLAYER_VIP                               =210;//
	SW2C_SET_PLAYER_VIP_RET                           =211;//

	SW2C_RECREATE_PAY_ORDER                           =212;//
	SW2C_RECREATE_PAY_ORDER_RET                       =213;//

	SW2C_SEND_SERVERS_NOTICE                          =214;//
	SW2C_SEND_SERVERS_NOTICE_RET                      =215;//

	SW2C_SERVERS_HOT_RES_RELOAD                       =216;//
	SW2C_SERVERS_HOT_RES_RELOAD_RET                   =217;//

	SW2C_CREATE_GIFT_CODE                             =218;//
	SW2C_CREATE_GIFT_CODE_RET                         =219;//

	SW2C_GET_ONLINE_PLAYER_COUNT                      =220;//
	SW2C_GET_ONLINE_PLAYER_COUNT_RET                  =221;//


	SW2C_GET_SERVER_INFO                              =222;//
	SW2C_GET_SERVER_INFO_RET                          =223;//

	SG2D_PVP_MARK_FIGHT                               =224;
	SG2D_PVP_MARK_FIGHT_RET                           =225;

////////////////////////////////////////////////////////////////////////////////
//副本
	SG2D_UPDATA_FUBEN_BASE                            =226;
	SG2D_UPDATA_FUBEN_RESOURCE                        =227;
	SG2D_UPDATA_FUBEN_PASSSTAGE                       =228;

////////////////////////////////////////////////////////////////////////////////
//国家(服务器的)
	SG2D_COUNTRY_BASE                                 =229;
	SG2D_COUNTRY_ADD_CITY                             =230;
	SG2D_COUNTRY_DEL_CITY                             =231;
	SG2D_COUNTRY_ALL_INFO                             =232;
	SD2G_COUNTRY_ALL_INFO                             =233;
	SG2D_COUNTRY_LEVLE_UP_TASK_RANK                   =234;
	SD2G_COUNTRY_LEVLE_UP_TASK_RANK                   =235;

////////////////////////////////////////////////////////////////////////////////
//市集、地牢、祭坛
	SG2D_MARKET_COLLECT                               =236;
	SG2D_MARKET_BLACK                                 =237;
	SG2D_MARKET_BUY                                   =238;
	SG2D_MARKET_MYSTERY                               =239;
	SG2D_ALTAR_ALL_INFO                               =240;
	SG2D_DUNGEON_BASE_INFO                            =241;
	SG2D_DUNGEON_ADD_HERO                             =242;
	SG2D_DUNGEON_DEL_HERO                             =243;
	SG2D_BAO_QI_ALL_INFO                              =244;

//铁匠铺
	SG2D_UPDATE_SMITHY                                =245;

////////////////////////////////////////////////////////////////////////////////
//皇城官职
	SG2D_PALACE_ALL_INFO                              =246;
	SD2G_PALACE_ALL_INFO                              =247;
	SG2D_PALACE_ACHIEVEMENT_INFO                      =248;
	SD2G_PALACE_ACHIEVEMENT_INFO                      =249;

////////////////////////////////////////////////////////////////////////////////
//PlayerMapLogic
	SG2D_MAP_LOGIC_BASE                               =250;
	SG2D_MAP_LOGIC_UPDATE_CITY_EVENT                  =251;
	SG2D_MAP_FOG_REC                                  =252;

////////////////////////////////////////////////////////////////////////////////
//科技
	SG2D_UPDATA_TECHNOLOGY_STUDY                      =253;

//跨服战
	SG2R_REGIST_CROSS_PLAYER_INFO                     =254;
	SB2G_GET_COUNTRY_INFO                             =255;
	SB2G_GET_COUNTRY_INFO_RET                         =256;
	SB2G_UPDATE_CROSS_WAR_INFO                        =257;

//远征
	SG2D_CRUSADE_UPDATE                               =258;

//每日任务
	SG2D_DAILY_QUEST_UPDATE                           =259;
//成长任务
	SG2D_GROW_UP_QUEST_UPDATE                         =260;
	SG2D_GROW_UP_QUEST_GROUP_UPDATE                   =261;


//签到
	SG2D_SIGN_UP_UPDATE                               =262;
////通关送钻
	SG2D_ACTIVITY_PASS_STAGE_REWARD_DIAMOND_UPDATE    =263;
/////ACTIVITY
	SG2D_ACTIVITY_STAR_BOX_UPDATE                     =264;
	SG2D_ACTIVITY_STAR_BOX_DELETE                     =265;

//后台控制活动
	SC2G_UPDATE_ACTVIITY_DATA                         =266;
	SC2G_UPDATE_ACTVIITY_DATA_RET                     =267;
	SG2D_UPDATE_ACTIIVTY_DATA                         =268;
	SG2D_UPDATE_ACTIVITY_DATA_RET                     =269;
	SD2G_UPDATE_ACTIVITY_DATA                         =270;
	SD2G_UPDATE_ACTIVITY_DATA_RET                     =271;

	SD2G_LOAD_ACTIIVTY_INFO                           =272;
	SD2G_LOAD_ACTIIVTY_INFO_RET                       =273;

	SG2D_GROW_UP_QUEST_DELETE_INFO                    =274;
	SG2D_PASS_STAGE_REWARD_DIAMOND_DELETE             =275;

///zhengwu
	SG2D_ZHENGWU_UPDATE                               =276;

//Generalreward
	SG2D_ACTIVITY_GENERAL_REWARD_UPDATE               =277;
	SG2D_ACTIVITY_GENERAL_REWARD_DELETE               =278;
////////worldfightachievement
	SG2D_WORLD_FIGHT_ACHIEVEMENT_UPDATE               =279;

///////countryselect
	SG2D_SELECT_COUNTRY                               =280;
	SD2G_SELECT_COUNTRY_RET                           =281;

////玩家加载完
	SD2G_PLAYER_HAS_ALL_SYNC                          =282;

//攻城部队
	SG2D_SEIGE_FORCE_DATA_UPDATE                      =283;

///
	SG2D_SAVE_ALL_PLAYER_CACHE                        =284;
/////连城
	SG2D_CONTINUE_OCCUPY_CITY_UPDATE                  =285;
///
	SD2G_ACTIVITY_COUNTRY_OCCUPY_CITY                 =286;
	SG2D_ACTIVITY_COUNTRY_OCCUPY_CITY_UPDATE          =287;

///恭贺功能
	SG2D_CONGRATULATE_UPDATE                          =288;
	SG2D_CONGRATULATE_BEEN_CON_UPTATE                 =289;

///////Wannabestronger
	SG2D_WANNA_BE_STRONGER_UPDATE                     =290;

//////BranchFuben
	SG2D_BRANCH_FUBEN_UPDATE                          =291;
	SG2D_HIDE_FUBEN_UPDATE                            =292;
////////////////////////////////////////////////////////////////////
//CrossWarsAbout
//                                                =300$;
	SL2S_NOTICE_WAR_STATE                             =300;//lobby->server通知各个服务器跨服阶段

	SL2G_NOTICE_WAR_START                             =301;//lobby->game跨服灭国开始
	SL2R_NOTICE_WAR_START                             =302;//lobby->cross跨服灭国开始

	SL2G_COLLECT_COUNTRY_INFO                         =303;//lobby->game收集所有可参战国家信息
	SG2L_COLLECT_COUNTRY_INFO_RET                     =304;//game->lobby返回信息

	SL2G_DEVIDE_CROSS_SERVER                          =305;//发送给对应的游戏服的国家
	SL2R_DEVIDE_CROSS_SERVER                          =306;//发送给跨服服务器分配的国家信息
	SG2R_UPDATE_PLAYER_VALUE                          =307;//游戏服务器与跨服服务器之间同步玩家属性
	SR2G_QUIT_CROSS_SERVER                            =308;

	SG2R_OFFICE_INFO                                  =309;//官职信息

	SR2G_Cross_Result_Country                         =310;//跨服战奖励：国家

	SG2R_UPDATE_PLAYER_CHAR_VALUE                     =311;//武将经验值增量

	SR2G_NOTICE_CLOSE_CROSS                           =312;//通知游戏服务器处理完毕，准备关闭跨服

	SR2G_Change_Awards                                =313;//跨服奖励发送给游戏服

	SG2R_Check_Can_Change_Hero                        =314;//检测能否换武将

	SG2B_Ask_Cross_Info                               =315;//重启时请求跨服信息
	SB2G_Ask_Cross_Info_Ret                           =316;

	SR2G_Award_Take_Player                            =317;

///////////////////////////////////////////////////////////////////
//世界事件
//                                                =350$;
	SG2D_UpdatePersonalEC                             =350;

	SERVER_OPCODE_COUNT                               =351;

}




